### ISSUE TITLE
When creating a new vocabulary, Simplify throws warning

### ISSUE THREAD

**Issue by argiepiano** [backdrop-contrib/simplify#8]
When creating a new vocabulary with Simplify enabled, the following is thrown:

`Warning: Illegal offset type in form_type_checkboxes_value() (line 2449 of /Users/XXX/Sites/localhost/bdtesting/core/includes/form.inc).`

The issue is line 139 of `simplify_form_taxonomy_form_vocabulary_alter()`. Since the vocab is new, the machine name is null, and this causes the whole array structure stored in `simplify.taxonomy.json` to be output.

**Comment by argiepiano** [comment]
PR #8 submitted.

In addition to checking if the vocabulary is new, I'm also doing an "array_filter" for default_value. It doesn't make sense to pass a bunch of `0` values in the default_value array for checkboxes that are not enabled. The _values_ of the default_value array are used to change the state of the checkboxes; those value are the keys of the checkbox that are supposed to be enabled. The keys of the default_value array are simply ignored. Therefore  passing an array of `0` values keyed by the names of the checkboxes has no effect and is not needed.    In other words, the default_value array for multiple checkboxes should be an indexed unidimensional array (not an associative array).

**Comment by herbdool** [comment]
Might need to get someone to test it and wait until the Bug Squad can merge since the maintainer is no longer active.

**Comment by yorkshire-pudding** [comment]
@argiepiano @herbdool 
I've tested this and can confirm it works.  Also, to note that in PHP 8.1, this is an Error and stops the page loading

There is a [related issue for content types](10) and I've added a PR ( #11) to fix that; I've used a similar approach to what @argiepiano used here, but the properties of the object are different, config loaded for the type and for comments, so I've done it slightly different.

It would be good to get both these merged and released ASAP because it is a tricky issue for users to work out that this module is causing it.
